### DISCUSSION TITLE
CSS semicolons getting removed

### DISCUSSION THREAD

**Discussion by mbowles-bhill** [mganss/HtmlSanitizer#560] [Q&A]
My application contains Rich Text boxes that allow users to add styles to their text. I am attempting to Sanitize the content and compare the sanitized input to the original to find if any malicious html was inserted. However, some css styles include semicolons in the html, but the sanitizer removes them, so I get a false positive on malicious html. For example underlined text comes in as 
![image](https://github.com/user-attachments/assets/8aec3b1d-e393-4c78-b707-00287101ee8f)
but the sanitizer removes the semicolons. But because they are now different the program thinks that there was malicious html because the sanitizer stripped the semicolon out.
![image](https://github.com/user-attachments/assets/87326480-2ea7-4f7d-a392-ed29d41109d0)

Is there something I can configure so the semicolon is not removed?

**Comment by tiesont** [comment]
It's not really HtmlSanitier removing those characters, but rather AngleSharp - it attempts to normalize any CSS it processes. Not sure if there's an option to prevent what you're seeing.

**Reply by mbowles-bhill** [reply]
Yea I came to the same conclusion later on. I ended up changing my code to operate off of the remove events rather than compare the sanitized input. I figure if a Remove event fired then there was something not allowed in the content. Takes care of my issue but still annoying that AngleSharp is doing that
